src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -749,7 +749,21 @@ private Boolean IsOpenable(Boolean needToUncompress, Boolean needToLoadIntoMemor
                     if (CompressionMethod != CompressionMethodValues.Stored &&
                         CompressionMethod != CompressionMethodValues.Deflate)
                     {
-                        message = SR.UnsupportedCompression;
+                        switch (CompressionMethod)
+                        {
+                            case CompressionMethodValues.Deflate64:
+                                message = SR.UnsupportedCompression + " Deflate64";
+                                break;
+                            case CompressionMethodValues.BZip2:
+                                message = SR.UnsupportedCompression + " Bzip2";
+                                break;
+                            case CompressionMethodValues.LZMA:
+                                message = SR.UnsupportedCompression + " LZMA";
+                                break;
+                            default:
+                                message = SR.UnsupportedCompression;
```

Comment:
Should the compression method be part of the resource string as "{0}", with it then filled in with SR.Format, e.g.
```C#
message = SR.Format(SR.UnsupportedCompression, CompressionMethod.ToString());
```
?